pkg/model/bootstrapscript.go

```diff
@@ -88,3 +94,67 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 	}
 	return fi.WrapResource(templateResource), nil
 }
+
+func (b *BootstrapScript) createProxyEnv(ps *kops.EgressProxySpec) string {
+	var buffer bytes.Buffer
+
+	if ps != nil && ps.HTTPProxy.Host != "" {
+		var httpProxyUrl string
+
+		// TODO double check that all the code does this
+		// TODO move this into a validate so we can enforce the string syntax
+		if !strings.HasPrefix(httpProxyUrl, "http://") {
+			httpProxyUrl = "http://"
```

Comment:
Good catch.  There's something else here that is more subtle... the egressProxy specifies a "Host" not a "URL".  This means the transport is not really expected to be specified in the host string.  In most cases, the transport from client to proxy is http, so for this PR this is what we assume, though it is possible in some situations to handle encryption from client to proxy, via https.  To support this feature, we would just add an api for HTTPProxy.Transport.  Meanwhile, what I think we wanted here was for `httpProxyUrl` to be `ps.HTTPProxy.Host`, I'll make that change now.